presto-feature-toggle/README.md

```diff
@@ -0,0 +1,233 @@
+
+## Contents
+
+1. [Configuration](#configuration)
+2. [Defining Feature Toggles](#defining-feature-toggles)
+    1. [Simple feature toggle definition](#simple-feature-toggle-definition)
+    2. [Hot reloadable feature toggle definition](#hot-reloadable-feature-toggle-definition)
+    3. [Strategy-based Feature Toggling](#strategy-based-feature-toggling)
+3. [Examples](#examples)
+    1. [Query Cancel Feature](#query-cancel-feature)
+    2. [feature-config.properties file example](#feature-toggle-config-file-example)
+
+## Configuration
+
+To allow feature toggle configuration four lines are needed in the config.properties file
+
+```
+    features.config-source-type=file
+    features.config-source=/etc/feature-config.properties
+    features.config-type=properties
+    features.refresh-period=30s
+```
+
+- `configuration-source-type` is the source type for Feature Toggles configuration
+- `features.config-source` is a source (file) of the configuration
+- `features.config-type` format in which configuration is stored (JSON or properties)
+- `features.refresh-period` configuration refresh period
+
+## Defining Feature Toggles
+
+Feature toggle definition is done in the Google guice module using `FeatureToggleBinder`
+
+## Simple feature toggle definition
+
+```
+    featureToggleBinder(binder)
+        .featureId("featureXX")
+        .bind()
+```
+
+This example creates bindings for `@FeatureToggle("featureXX") Supplier<Boolean> isFeatureXXEnabled.`
+
+```   
+    @Inject
+    public Runner(@FeatureToggle("featureXX") Supplier<Boolean> isFeatureXXEnabled)
+    {
+        this.isFeatureXXEnabled = isFeatureXXEnabled;
+    }
+```
+
+Supplier&lt;Boolean&gt; `isFeatureXXEnabled` can be used to test if the feature is enabled or disabled:
+
+```
+    boolean testFeatureXXEnabled()
+    {
+      return isFeatureXXEnabled.get();
+    }
+```
+
+Switching the feature toggle on/off is done by changing the enabled value from true to false in the configuration source file:
+
+```
+    feature.featureXX.enabled=true
+```
+
+After the refresh period value of `isFeatureXXEnabled.get();`  is changed.
+
+## Hot reloadable feature toggle definition
+
+```
+    featureToggleBinder(binder, Feature01.class)
+        .featureId("feature01")
+        .baseClass(Feature01.class)
+        .defaultClass(Feature01Impl01.class)
+        .allOf(Feature01Impl01.class, Feature01Impl02.class)
+        .bind()
+```
+
+```
+    class Runner
+    {
+        private final Provider<Feature01> feature01;
+    
+        @Inject
+        public Runner(@FeatureToggle("feature01") Provider<Feature01> feature01)
+        {
+            this.feature01 = feature01;
+        }
+
+        public String testFeature01()
+        {
+            return feature01.get().test();
+        }
+    }
+```
+
+## Strategy-based Feature Toggling
+
+Strategy-based Toggling allows us to Implement custom predicates (Strategy Pattern) to evaluate if a feature is enabled.
```

Comment:
```suggestion
Strategy-based Toggling allows us to implement custom predicates (Strategy Pattern) to evaluate if a feature is enabled.
```
Can you rephrase to remove the parentheses around "Strategy Pattern"? Also, if Strategy Pattern is an important concept here, consider adding a link to help people learn. Maybe the wikipedia entry is appropriate? 
https://en.wikipedia.org/wiki/Strategy_pattern